gossip_test.go

```diff
@@ -22,7 +24,7 @@ var _ gossipConnection = &mockGossipConnection{}
 
 func newTestRouter(name string) *Router {
 	peerName, _ := PeerNameFromString(name)
-	router := NewRouter(Config{}, peerName, "nick", nil)
+	router := NewRouter(Config{}, peerName, "nick", nil, log.New(ioutil.Discard, "", 0))
```

Comment:
I agree with @peterbourgon's general thrust, but would add that at step 2 one typically wants a logger with debug-level turned on.

Maybe a package-scope variable for the logger would be better than a locally-created one you have to edit?

I can't think of any cases where we were able to debug a rare test failure specifically from the non-debug logs.  Maybe from the panic trace - that reminds me we might want to override Go 1.6's change to printing just the current goroutine.